packages/cardpay-sdk/sdk/revenue-pool/base.ts

```diff
@@ -97,22 +97,36 @@ export default class RevenuePool {
     return gasInToken;
   }
 
+  async claim(txnHash: string): Promise<TransactionReceipt>;
   async claim(
     merchantSafeAddress: string,
     tokenAddress: string,
     amount: string,
     txnOptions?: TransactionOptions,
     contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt>;
+  async claim(
+    merchantSafeAddressOrTxnHash: string,
+    tokenAddress?: string,
+    amount?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
   ): Promise<TransactionReceipt> {
+    if (!tokenAddress) {
+      let txnHash = merchantSafeAddressOrTxnHash;
+      return waitUntilTransactionMined(this.layer2Web3, txnHash);
+    }
+    let merchantSafeAddress = merchantSafeAddressOrTxnHash;
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
     let from = contractOptions?.from ?? (await this.layer2Web3.eth.getAccounts())[0];
     let revenuePoolAddress = await getAddress('revenuePool', this.layer2Web3);
     let revenuePool = new this.layer2Web3.eth.Contract(RevenuePoolABI as AbiItem[], revenuePoolAddress);
     let unclaimedBalance = new BN(await revenuePool.methods.revenueBalance(merchantSafeAddress, tokenAddress).call());
-    if (unclaimedBalance.lt(new BN(amount))) {
+    if (unclaimedBalance.lt(new BN(amount!))) {
```

Comment:
ditto on comment about just checking for `amount == null` and throwing so we don't have to use `!`'s